AutoSteamApp/Process Memory/SteamworksData.cs

```diff
@@ -0,0 +1,235 @@
+            Log.Debug("2nd Phase Address: " + SecondPhaseAddress);
+            // Offset to find the rarity
+            RarityAddress = SteamworksAddress + MHWMemoryValues.OffsetToGameRarity;
+            Log.Debug("Rarity Address: " + RarityAddress);
+            SteamGaugeAddress = SteamworksAddress + MHWMemoryValues.OffsetToSteamGauge;
+            Log.Debug("Steam Gauge Address: " + SteamGaugeAddress);
+        }
+
+        /// <summary>
+        /// Returns a tuple of keycodes to press corresponding to the correct sequence of moves .
+        /// </summary>
+        /// <returns>A Tuple </returns>
+        public VirtualKeyCode[] ExtractSequence()
+        {
+            // Read the byte representation of the sequence from the game
+            byte[] sequence = new byte[3];
+            sequence[0] = MemoryHelper.Read<byte>(MHWProcess, SequenceAddress);
+            sequence[1] = MemoryHelper.Read<byte>(MHWProcess, SequenceAddress + 1);
+            sequence[2] = MemoryHelper.Read<byte>(MHWProcess, SequenceAddress + 2);
+
+            // return null if they are are all zero (indicates the minigame hasnt started)
+            if (sequence.AllIdentical())
+                if (sequence[0] == 0)
+                    return null;
+                else
+                    throw new Exception("Error interpretting sequence [" + string.Join(", ", sequence.Select(x => x.ToString())) + "]");
+
+            // Capcom has likely done some manipulation so we need to do some twidling because the sequences aren't as straightforward as we'd like.
+            // Kudos to https://github.com/Geobryn for figuring this out,
+            if (sequence[0] == 2 && sequence[1] == 0 && sequence[2] == 1)
+            {
+                sequence[0] = 1;
+                sequence[1] = 2;
+                sequence[2] = 0;
+            }
+            else
+                if (sequence[0] == 1 && sequence[1] == 2 && sequence[2] == 0)
+            {
+                sequence[0] = 2;
+                sequence[1] = 0;
+                sequence[2] = 1;
+            }
+
+            // Return the array of key code presses extracted from the bytes
+            return StaticHelpers.KeyCodeSequenceFromBytes(sequence);
+        }
+
+        #endregion
+
+    }
+
+    /// <summary>
+    /// Enum used for determining which state of button press are we in
+    /// </summary>
+    public enum ButtonPressedState
+    {
+        Beginning = 0,
+        End = 7
+    }
+
+    /// <summary>
+    /// Enum used for determining which state of the steamworks animations are we in
+    /// </summary>
+    public enum PhaseState
+    {
+        Idle = 0,
+        Burning = 1,
+        Fuel = 2,
+        Bonus = 4,
+        Settled = 5,
+        WaitingForInput = 8,
+        Cutscene = 12,
+        Rewards = 13,
+    }
+
+    /// <summary>
+    /// The rarity of the reward of the steamworks
+    /// </summary>
+    public enum RewardRarity
+    {
+        Common = 0,
+        Rare = 1
```

Comment:
Any reference for this value? I haven't manage to find it.

Reply:
I'm using this in ExtractAndEnterSequence. Here's an excerpt
`if (_SteamworksData.RewardRarityValue == RewardRarity.Rare)
                {
                    Log.Debug("Rare reward detected.");
                    probability = ConfigurationReader.RareSuccessRate;
                }`

It's actually the original reason I wanted to get started on this project. I felt that always 100% was too "cheaty" for me, but I still wanted to get the rare rewards haha.